File: Part2/C++/BasicConcepts/BasicConcepts_MixedTypes.cpp
#include <cstdio>
#include <climits>
#include <cfloat>

int main()
{
    // Do not mix your types in many languages
    int myInt = 123.45;
    float myFloat = 'a';
    char myChar = 543.21;
    
    printf ( "myInt: %d\n", myInt );
    printf ( "myFloat: %f\n", myFloat );
    printf ( "myChar: %c\n", myChar );
    
    return 0;
}

File: Part2/Fortran/BasicConcepts/BasicConcepts_MixedTypes.f90
program MixedTypes
    implicit none
    
    ! Do not mix your types in many languages
    ! Fortran is strongly typed and will not allow implicit type conversions
    ! These will cause compile errors due to type mismatches
    integer :: myInt = 123.45
    real :: myFloat = 'a'
    character(len=1) :: myChar = 543.21
    
    write(*,*) 'myInt:', myInt
    write(*,*) 'myFloat:', myFloat
    write(*,*) 'myChar:', myChar

end program MixedTypes
